openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -267,13 +269,65 @@ def _does_name_change_require_verification(user, old_name, new_name):
     """
     If name change requires verification, do not update it through this API.
     """
+
+    user_profile = UserProfile.objects.get(user=user)
+    profile_meta = user_profile.get_meta()
+    name_changes = 0
+    if 'old_names' in profile_meta:
+        name_changes = len(profile_meta['old_names'])
+
     return (
         is_verified_name_enabled()
-        and old_name != new_name
-        and len(get_certificates_for_user(user.username)) > 0
+        and (
+            _edits_are_invalid(old_name, new_name)
+            or name_changes >= 2
+            or len(get_certificates_for_user(user.username)) > 0
+        )
     )
 
 
+def _edits_are_invalid(old_name, new_name):
+    """
+    Check if the edits from old name to new name are invalid
+
+    Edits are considered invalid if:
+    * Two or more spaces occur in a row
+    * More than one non-space character is added/removed/replaced
+    * Only one space can be added alone, or one space can be added on either side of a new character
+    """
+    # if there are multiple spaces in a row, it is invalid
+    contains_multiple_spaces = bool(re.search(r' {2,}', new_name))
+    if contains_multiple_spaces:
+        return True
+
+    modifications = 0
+
+    # get differences between old name and new name
+    sequence = SequenceMatcher(lambda x: x == " ", old_name, new_name)
+    for tag, i1, i2, j1, j2 in sequence.get_opcodes():
+        # if there is more than one sequence in the string that has been modified, edits are invalid
+        if modifications > 1:
+            return True
+
+        # if tag is anything other than equal, increase modifications
+        if tag != 'equal':
+            modifications += 1
+
+            # determine which piece has been modified
+            old_name_substring = old_name[i1:i2]
+            new_name_substring = new_name[j1:j2]
+            modified_substring = (
+                old_name_substring
+                if len(old_name_substring) > len(new_name_substring)
+                else new_name_substring
+            )
+            is_valid = bool(re.search(r'^\s?(\S\s?)?$', modified_substring))
```

Comment:
I wasn't sure how many limitations we should put on what's considered a character, which is why I went with `\S`. Prior to checking if the name is valid for name affirmation, we check if the name contains any [html or urls](https://github.com/edx/edx-platform/blob/1578b0c6ea07330906a042566cd8bffaa96efdaa/openedx/core/djangoapps/user_authn/views/registration_form.py#L100-L109), so I'm not sure if further checks are needed here.